pkg/generate/app/cmd/newapp.go

```diff
@@ -418,6 +418,19 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						return nil, err
 					}
 				}
+				// Append any exposed ports from Dockerfile to input image
```

Comment:
> Port detection was previously happening while in source detection (removed by this PR). We should restore it somewhere there and not so late in the new-app flow.

An issue describing what we'd like to achieve in a refactor would be sufficient for me.  I tried to make this modification myself, but after spending an hour in the code, I couldn't figure out where to do it.  My changes kept getting stomped.